File: regalaya-web/src/services/payment.service.ts
/**
 * Payment Service
 *
 * Serviço para operações de pagamento (PIX, Cartão de Crédito).
 */

import { http } from '@/lib/api'
import type {
  PaymentIntentRequest,
  PaymentIntentResponse,
  PaymentStatusResponse,
  InstallmentOption,
} from '@/types/payment'

export const paymentService = {
  /**
   * Cria um payment intent para um pedido
   */
  createIntent: (request: PaymentIntentRequest) =>
    http.post<PaymentIntentResponse>('/payments/create-intent', request),

  /**
   * Obtém o status de pagamento de um pedido
   */
  getPaymentStatus: (orderId: string) =>
    http.get<PaymentStatusResponse>(`/payments/${orderId}`),

  /**
   * Calcula opções de parcelamento
   */
  calculateInstallments: (total: number, maxInstallments = 12) =>
    http.get<InstallmentOption[]>('/payments/installments', { total, maxInstallments }),
}

File: regalaya-api/src/main/java/br/com/regalaya/payment/services/PaymentService.java
package br.com.regalaya.payment.services;

import br.com.regalaya.payment.dto.requests.PaymentIntentRequest;
import br.com.regalaya.payment.dto.responses.InstallmentOption;
import br.com.regalaya.payment.dto.responses.PaymentIntentResponse;
import br.com.regalaya.payment.dto.responses.PaymentStatusResponse;

import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

public interface PaymentService {

    /**
     * Creates a payment intent for the given order
     */
    PaymentIntentResponse createPaymentIntent(UUID userId, PaymentIntentRequest request);

    /**
     * Gets payment status for an order
     */
    PaymentStatusResponse getPaymentStatus(UUID orderId);

    /**
     * Calculates installment options for a given total
     */
    List<InstallmentOption> calculateInstallments(BigDecimal total);

    /**
     * Handles webhook notification from payment provider
     */
    void handleWebhook(String provider, String eventType, String providerEventId, String payload);

    /**
     * Processes expired PIX payments
     */
    int processExpiredPixPayments();
}
